### PR TITLE
Fix: restore launch flag --use-app-center

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

launch flag `--use-app-center` not work

### Causes

The flag is removed after App center intergration

### Solutions

Remove old Hockey app related code and redo fro AppCenter.

App center is enabled or disabled with these checkings:
priority 1:
check launching option `--use-app-center`'s value (0/1)

priority 2:
check UserDefaults KVP `UseAppCenter`

priority 3:
check Bundled info plist has `UseAppCenter` (form .xcconfig key `USE_APP_CENTER_KEY`)

### CODE REVIEW

File: WireCommonComponents/AutomationHelper.swift [hunk 1]
```diff
@@ -89,14 +96,24 @@ public final class AutomationHelper: NSObject {
...
+        
+        let value = arguments.flagValueIfPresent(AutomationKey.useAppCenter.rawValue)
+        switch value {
+        case "0":
+            useAppCenterLaunchOption = false
+        case "1":
+            useAppCenterLaunchOption = true
+        default:
+            break
+        }
```
Comment (johnxnguyen) [inline WireCommonComponents/AutomationHelper.swift:114]: Why do we set the `useAppCenterLaunchOption` differently from the rest?

Comment (billypchan) [reply WireCommonComponents/AutomationHelper.swift:114]: There is an existing `useAppCenter` property which gets a stored value from `UserDefaults`. The new `useAppCenterLaunchOption` property has a higher priority than that.

**Review by johnxnguyen — COMMENTED** [review]
@billypchan can you describe what the intended behavior should be because it's a bit confusing from the code.

**Comment by billypchan** [comment]
@johnxnguyen updated the description to explain the user app center flag in launch option, user default and bundled plist

**Review by johnxnguyen — COMMENTED** [review]
So we only care about negative values? The behavior I see is this:

```swift
let useAppCenter = launchArgumentValue && userDefaultsValue && bundleValue
```

Which means that the launch argument is `--use-app-center` is `true`, but the user defaults value is `false`, then app center would be disabled. Is this correct?

**Comment by billypchan** [comment]
@johnxnguyen I will clean up the code and put all logic in one place.

File: Wire-iOS/Sources/AppDelegate+AppCenter.swift [hunk 2]
```diff
@@ -32,9 +32,10 @@ extension AppDelegate {
 
     func setupAppCenter(completion: @escaping () -> Void) {
 
-        let shouldUseAppCenter = AutomationHelper.sharedHelper.useAppCenter || Bundle.useAppCenter
+        // launch option has higher priority than bundle option
+        let shouldUseAppCenter = AutomationHelper.sharedHelper.useAppCenter && Bundle.useAppCenter
```
Comment (billypchan) [inline Wire-iOS/Sources/AppDelegate+AppCenter.swift:36]: @johnxnguyen `AutomationHelper.sharedHelper.useAppCenter` here has `launchArgumentValue && userDefaultsValue` check inside

**Comment by johnxnguyen** [comment]
> @johnxnguyen I will clean up the code and put all logic in one place.

I was asking a question about what the flags mean. It's not clear at all what effect a single flag is supposed to have. Is each flag really only about whether we should disabled app center?

**Comment by billypchan** [comment]
@johnxnguyen the flags are coming from different requirements:

`launchArgumentValue` - only for QA, when running the app with automation
`userDefaultsValue` - for manually control app center with user default, for testing
`bundleValue` - build time force disable depends on config.

For the question "Which means that the launch argument is --use-app-center is true, but the user defaults value is false, then app center would be disabled. Is this correct?"

The use case of `--use-app-center` for QA is set it to false to force disable app center. Set it to ture is equal to not to set it and will let other 2 booleans make the decision.

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**